zerver/actions/realm_emoji.py

```diff
@@ -49,7 +57,7 @@ def check_add_realm_emoji(
     realm_emoji.is_animated = is_animated
     realm_emoji.save(update_fields=["file_name", "is_animated"])
 
-    realm_emoji_dict = realm.get_emoji()
+    realm_emoji_dict = get_realm_emoji_by_id(realm.id)
```

Comment:
Can we do `get_all_realm_emoji_by_id(realm_id)` or `get_custom_emoji_for_realm` for this name? Read this, I think we're looking up a single emoji by `realm_emoji.id`, sigh.

Reply:
I'll try out some better names.  I'm probably gonna squash the prep commit here into the commit that unifies the cache, since you've already taken a review pass, and since there's no really good intermediate state of the refactoring.